scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java

```diff
@@ -433,19 +418,35 @@ public JobId submit(Job userJob)
                 throw new SubmissionClosedException(msg);
             }
 
-            UserIdentificationImpl ident = frontendState.checkPermission("submit",
-                                                                         YOU_DO_NOT_HAVE_PERMISSION_TO_SUBMIT_A_JOB);
+            final UserIdentificationImpl ident = frontendState.checkPermission("submit",
+                                                                               YOU_DO_NOT_HAVE_PERMISSION_TO_SUBMIT_A_JOB);
 
-            InternalJob job = frontendState.createJob(userJob, ident);
+            final InternalJob job = frontendState.createJob(userJob, ident);
 
-            schedulingService.submitJob(job);
+            pool.submit(() -> {
+                try {
+                    schedulingService.submitJob(job);
+                    frontendState.jobSubmitted(job, ident);
+                } catch (Exception e) {
+                    logger.error(String.format("Error while submiting a job[id:%s] ", job.getId().value()), e);
+                } finally {
+                    unregisterSumbittedJob(job.getId());
+                }
 
-            frontendState.jobSubmitted(job, ident);
+            });
+            while (job.getId().longValue() == 0) {
+                try {
+                    Thread.sleep(10);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)